### DISCUSSION TITLE
Track your grocery spending and consumption and observe price trends

### DISCUSSION THREAD

**Discussion by djyotta** [sqlpage/SQLPage#319] [Show and tell]
I've been tracking our household grocery purchases for my own interest for a few years now.

The db (postgresql) is the first class UI.

I wrote a web site for use on mobile device and to plot charts. But I'm aiming for the db to be human workable and do all the "logic" via appropriate constraints and sensible relations.

I recently converted my web UI to SQLPage UI instead of the Python + Bottle + XSLT I had earlier:

https://gogs.shandan.one/gogsadmin/grocery-manager/src/sqlpage

It's a much cleaner and better looking interface:

https://shandan.one/grocery/trend.sql

When I found out about SQLPage, I wanted to convert my UI as it was exactly the spirit of what I was looking for... database driven UI.

Whilst my dream was for super thin (technically speaking), web layer to interface the db (ie, just tables coming out), no JS or other cruft, I find SQLPage a much closer fit to this philosophy than rolling my own.

The source code (SQL) of my web site seems thinner now, with SQLPage doing the heavy lifting. I don't mind the JS so long as I don't have to maintain it.

The charts SQLPage provides are nicer than what I got after striving with matplotlib, don't take a long time to generate and don't require my own caching mechanism.

Overall a significant improvement and most importantly simplification to move to SQLPage for my DB driven website.

**Comment by lovasoa** [comment]
This is a great app ! Pasting a link to the source code here : https://gogs.shandan.one/gogsadmin/grocery-manager/src/sqlpage

And mentioning a few people who may be interested: @DSMejantel , @3n-mb, @matthewlarkin , @olivierauverlot 

![image](https://github.com/lovasoa/SQLpage/assets/552629/2bb93b77-9363-4059-b058-0354db8a59b0)
